### DISCUSSION TITLE
GeoJsonLayer: significant performance impact

### DISCUSSION THREAD

**Discussion by bortomar-intellmaps** [visgl/deck.gl#8185] [Q&A] [answered]
We notice that when we are trying to use layer's accessors getLineWidth and getColor for styling purpose of a "large" GeoJsonLayer (over 50k features) there is significant performance drop (even if we are using only constants). 
Is there any option how to improve performance?

## without accessors
[geojson-no-style.webm](https://github.com/visgl/deck.gl/assets/135999694/8bf24014-fc38-48bc-a1ec-0021c7902005)

## with accessors
```js
getLineColor: (f) => [0, 0, 0, 255],
getLineWidth: (f) => 0.05
```
[geojson-style.webm](https://github.com/visgl/deck.gl/assets/135999694/c20fd35f-a895-4654-8a3f-72cba8d2824e)

**Comment by Pessimistress** [comment, marked as answer]
https://deck.gl/docs/developer-guide/performance#favor-constants-over-callback-functions
